File: InsuranceDiscountsWeb/Core/Models/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Models
{
    public class Company
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public string CompanyName { get; set; } = String.Empty;

        [Required]
        public string Description { get; set; } = String.Empty;

        [Required]
        public string Address { get; set; } = String.Empty;

        [Required]
        [Phone]
        public string PhoneNumber { get; set; } = String.Empty;

        [Required]
        [EmailAddress]
        public string EmailAddress { get; set; } = String.Empty;

        [Required]
        [Url]
        public string WebsiteAddress { get; set; } = String.Empty;
    }
}

File: mobile_app/app/src/main/java/com/example/insurance_discounts/data/Company.kt
package com.example.insurance_discounts.data

data class Company(
    val id: String,
    val companyName: String,
    val description: String,
    val address: String,
    val phoneNumber: String,
    val emailAddress: String,
    val websiteAddress: String,
    val maxDiscountPercentage: Int
)